sparsesim/src/lib.rs

```diff
@@ -1074,6 +1074,8 @@ impl QuantumSim {
                         .fold(SparseState::default(), |mut accum, (index, value)| {
                             if ctls.iter().all(|c| index.bit(*c)) {
                                 accum.insert(index, value * factor);
+                            } else {
```

Comment:
Good catch, thanks for the fix! It looks like I made the same mistake on line 1091 below as well. Could you apply the same fix there too?

Reply:
Yes sir, just attempted that. What do you think?

Reply:
Looks great! Not sure why the pipeline is mad (looks like an intermittent issue with the license task, the tests are passing). I'll try to get this merged.